src/core/resolve.c

```diff
@@ -1623,7 +1623,12 @@ struct hostent *no_naptr_srv_sip_resolvehost(
 			srv_name.s = tmp_srv;
 			srv_name.len = strlen(tmp_srv);
 #ifdef USE_DNS_CACHE
-			he = dns_srv_get_he(&srv_name, port, dns_flags);
+			if(dns_cache_init) {
+				he = dns_srv_get_he(&srv_name, port, dns_flags);
+			} else {
+				LM_WARN("USE_DNS_CACHE is defined but dns_cache_init=off\n");
```

Comment:
Why is this a run-time warning, depending on how Kamailio is compiled and current settings.  I mean the distributions compile Kamailio and let the users select, `dns_cache_init` ON or OFF.  With this change users will get a warning, they do not to fix, as it will be up to the distribution to fix the warning, by preventing the users to toggle `dns_cache_init`.

Reply:
You are right, removed.